### ISSUE TITLE
Animation class in pygame

### ISSUE THREAD

**Issue by AndreyVarvar** [pygame-community/pygame-ce#2181]
i dont know if this was suggested before, but i think this is still a great thing to add in pygame.

Currently, you have to manually make a function for animations, so why not implement it in pygame? It is more convenient for a game framework.

My idea was to make an `Animation` class, that is initialized with a list of frames (`pygame.Surface()`-s) and a FPS, and the only thing left for the us, is to just call `Animation.update(dt)` (`dt` being time since last update).

In my opinion, this is a great addition to pygame-ce.

**Comment by yunline** [comment]
The Animation class may need to support both Surface and Texture. Or maybe `AnimationSurface` and `AnimationTexture` (?

**Comment by Matiiss** [comment]
I don't think this would be all that useful, there likely are libs out there that can provide you with this functionality and also everyone mostly does it their own way by now so yeah, I'm not really in support of this idea.

**Comment by durkisneer1** [comment]
This could be a method/attribute for the Sprite class. The same way you can draw a sprite as long as you have a "self.image" and "self.rect" attribute, I think a "self.animation" type list/tuple attribute with an "animate" method could work out.

**Comment by Starbuck5** [comment]
I’ve advocated before for an animation class outside of the sprite framework.

Perhaps it would be pygame.surface.AnimatedSurface, and blitting it would blit the current frame.

We’re going to need some sort of animation representation for when pygame.image.load supports GIFs.

**Comment by JiffyRob** [comment]
I like this idea a lot.  As for the "there are different libs" argument:
 - This would be implemented in C and therefore be faster than any third party python library.  I currently have a particle system set up with python implemented animations, and if it could be sped up that would be amazing.
 - Pygame already implements a lot of things that there are different libs for, including image manipulation, sound manipulation, midi interfacing, vector math, geometry, etc.
 - Speaking from experience, beginners don't like figuring out lots of libraries.  This would be very handy to have around, especially if `pygame.image.load` supports GIFs.
 - This is something that is a lot more common than, say, menu systems, or particle systems, or `.tmx` map loading.  At least it feels like to me it is used enough we shouldn't rely on other libs to do it properly.

Maybe I'm just rambling, but I would really like to see this happen.
